pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -105,17 +142,35 @@ func (m *sqlInstanceModel) AdapterForObject(ctx context.Context, kube client.Rea
 		sqlOperationsClient: gcpClient.sqlOperationsClient(),
 		sqlInstancesClient:  gcpClient.sqlInstancesClient(),
 		sqlUsersClient:      gcpClient.sqlUsersClient(),
+		statefulFields:      make(map[string]*StatefulField),
+	}
+
+	// Populate the map with all supported fields, initially marked as not unmanaged.
```

Comment:
Not sure there is a point in regenerating supportedUnmanageableFields. This seems static. Lets only generate this once.